package.json

```
@@ -166,7 +166,11 @@
       ]
     }
   },
+  "config": {
+    "pmdVersion": "7.0.0"
+  },
   "scripts": {
+    "postinstall": "cd pmd-packager && ./upgrade.sh $npm_package_config_pmdVersion",
```

Note: This requires now that you need to use git-bash if you are developing under Windows. postinstall will be executed every time "npm install" or "npm ci" is run.
See https://docs.npmjs.com/cli/v10/using-npm/scripts#life-cycle-operation-order

---

```suggestion
    "postinstall": "pmd-packager/upgrade.sh",
```

The env-var "npm_package_config_pmdVersion" should be also available inside the script, see https://docs.npmjs.com/cli/v10/using-npm/scripts#packagejson-vars